distributed/shuffle/_rechunk.py

```diff
@@ -127,32 +125,59 @@ class ShardID(NamedTuple):
     shard_index: NIndex
 
 
-def rechunk_slicing(
-    old: ChunkedAxes, new: ChunkedAxes
-) -> dict[NIndex, list[tuple[ShardID, NSlice]]]:
-    """Calculate how to slice the old chunks to create the new chunks
+class Split(NamedTuple):
+    """Slice of a chunk that is concatenated with other splits to create a new chunk
 
-    Returns
-    -------
-        Mapping of each old chunk to a list of tuples defining where each slice
-        of the old chunk belongs. Each tuple consists of the index
-        of the new chunk, the index of the slice within the composition of slices
-        creating the new chunk, and the slice to be applied to the old chunk.
+    Splits define how to slice an input chunk on a single axis into small pieces
+    that can be concatenated together with splits from other input chunks to create
+    output chunks of a rechunk operation.
     """
-    from dask.array.rechunk import intersect_chunks
 
-    ndim = len(old)
-    intersections = intersect_chunks(old, new)
-    new_indices = product(*(range(len(c)) for c in new))
+    #: Index of the new output chunk to which this split belongs.
+    chunk_index: int
+
+    #: Index of the split within the list of splits that are concatenated
+    #: to create the new chunk.
+    split_index: int
+
+    #: Slice of the input chunk.
+    slice: slice
+
 
-    slicing = defaultdict(list)
+SplitChunk: TypeAlias = list[Split]
+SplitAxis: TypeAlias = list[SplitChunk]
+SplitAxes: TypeAlias = list[SplitAxis]
```

Comment:
I don't have any better proposal right now but the difference between Ax**i**s and Ax**e**s doesn't help readability

Reply:
`SplitAxes` only pops up once so I guess this is not a big deal